fldigi_doxygen/user_src_docs/ConfigID.txt
/**
\page id_configuration_page  RSID / Video / ID Configuration

\tableofcontents

\image html config-id-rsid.png "RSID Config Panel"
\image latex config-id-rsid.png "RSID Config Panel" width=5.0in

Fldigi offers several ways to identify the operator or mode that is being
used. This is particularly useful when using a hard to recognize
mode such as \ref thor_page "Thor", \ref olivia_page "Olivia"
or \ref mt63_page "MT63".
<br>

\section subsec_reed_solomon_id Reed Solomon Identifier

<b>RSID</b>, <b>R</b>eed <b>S</b>olomon <b>ID</b>entifier
is a special transmission designed by Patrick Lindecker, F6CTE, for the
modem program MultiPsk. It has been adapted to other modem
programs. Fldigi's implementation is compatible with the MultiPsk
RSID, but provides a slight variation. You can transmit RSID at
both the beginning and end of a transmission. The detection of
RSID normally only occurs in the near vicinity of the current waterfall
tracking point. This cuts down on extraneous RSID detections when the
band is crowded and several RSID signals might be present. If you want
fldigi to search the entire waterfall for RSID signals you can do so by
enabling the "Detector searches entire passband". You start the
search for a signal based on RSID by using the main panel switch.
The RSID detector is a separate decoder that operates in parallel with all
other modem decoders. If you select the "Mark previous frequency and
mode" a restore link will be inserted into the RX text upon detecting
an RSID signal. Clicking on this link restores the previous
frequency and mode of operation. You elect to disable the RSID
upon first detection. You also have the option of just receiving
notification when an RSID signal is detected. The notification
occurs with a pop-up message box.
<br>

You can select which modes will include the transmitted RS identifier,
and which modes will react to a received and decoded RS identifier.
<br>

The mode to identifier relationships are selected by pressing the associated
"modes" button.

\image html config-id-rx-modes.png "Receive Modes"
\image latex config-id-rx-modes.png "Receive Modes" width=2.37in
<br>

\image html config-id-xmt-modes.png "Transmit Modes"
\image latex config-id-xmt-modes.png "Transmit Modes" width=2.37in

\section subsec_video_text Video Text
<br>

Transmitted video text will appear as a sequence of characters on the
waterfall.The text can be a brief mode identifier or some user specified
text. You can use a small font that always appears as a 2
character wide sequence or a larger font that can be 1 to 4 characters
wide. You should be aware that the video signal is a constant
energy signal and the content will be spread across multiple
characters. The highest s/n at the receiving end will be for 1
character wide video. Small font at 2 character width is next in
s/n performance followed by 2 character large font etc. You can
select which modes will include the video text preamble. You can
limit the horizontal (frequency width) of the video signal in one of
several inclusive ways.
<br>

<ul>
<li>Number of characters per row of text</li>
<li>Constrain to be less than or equal to 500 Hz</li>
<li>Constrain to be within the bandwidth limits of the mode in use</li>
</ul>

Fldigi uses abbreviated acronyms for the mode and it's characteristics
when you are transmitting the mode ID using a video text. Here
are two examples, one in small and the other in large font.<br>

\image html vid-small.png "Small Font"
\image latex vid-small.png "Small Font" width=1.2in
<br>

\image html vid-large.png "Large Font"
\image latex vid-large.png "Large Font" width=2.3in
<br>

Notice that Olivia 16-500 is abbreviated to OL-16/500 and that the
number of characters is limited to 8 per row. You might want to
use the large characters by default if you routinely have QSO's with
operators using older digital mode programs or one whose waterfall
visual is not on a par with fldigi's.
<br>

\image html config-id-video.png "Video ID Config Panel"
\image latex config-id-video.png "Video ID Config Panel" width=5.0in
<br>

\image html config-id-video-modes.png "Video ID Modes"
\image latex config-id-video-modes.png "Video ID Modes" width=1.8in

\section subsec_cw_postamble CW Postamble
<br>

You can transmit your callsign in CW as a postamble to all modes except of
CW (a bit redundant to do that). You can select which modes will
include the CW postamble.
<br>

<center>
\image html config-id-cw.png "CW Postamble ID Config Panel"
\image latex config-id-cw.png "CW Postamble ID Config Panel" width=5.0in
<br>

\image html config-id-cw-modes.png "CW ID Modes"
\image latex config-id-cw-modes.png "CW ID Modes" width=1.7in
</center>

<br>

\ref id_configuration_page "Return to Top of Page"
<br>
\ref main_page "Return to Main Page"

*/



fldigi_doxygen/user_src_docs/configCW.txt
/**
\page cw_configuration_page  CW Configuration

\tableofcontents
<br>
\image html config-cw-general.png "General CW Configuration"
\image latex config-cw-general.png "General CW Configuration" width=5.0in
<br>

Fldigi can send and receive morse code from 5 wpm to 200 wpm. The
operating controls for CW are found on the Config/CW tab. You can
open that tab by selecting the "Configure/Modems" menu item and the
clicking on the Modems/CW tab. You can also open up the CW tab by
first selecting CW as the operating mode and then clicking on the
left-most item "CW" on the status bar at the bottom of the fldigi main
window. During operation the Rx and Tx WPM settings are
annunciated on the status bar in the two boxes next to the mode
indicator.<br>

<br>The CW signals are converted to a baseband signal. It is the
digital equivalent of tuning an analog USB transceiver so that the
carrier is exactly at the CW carrier frequency. The CW decoder
can use one of two different DSP filters. The Fast Fourier
Transform (FFT) filter is implemented with a sin(x)/x
impulse response. This is a very steep sided low pass
filter. Unchecking the FFT filter causes the decoder to use a
Finite Impulse Response (FIR) filter with a cutoff slope that is not as
steep. The FFT filter is optimum when receiving CW in a white
noise environment. The FIR filter may give better response with
impulse noise. Selected a "Matched" filter for either the FFT or
FIR implementation optimizes the filter bandwidth for white noise
suppression. A lot of impuse noise (static) can cause either
filter to ring and increasing the filter bandwidth might improve
detection in that electrical environment.
<br>
<br>

Fldigi can track the incoming signal. Enable Rx WPM tracking by enabling
the check box "Enable Tx Trkg". The tracking range (+/- Hz around the
TxWPM setting) can be set using the "Rx Trkg Rng" control. When
tracking is enabled the tracking filter is reset every time the
transmit WPM is adjusted.
<br>
<br>

CW detection is basically an amplitude demodulator. You can set
the threshold for detecting when the signal transitions from off-to-on
and on-to-off. These signal levels are relative to the average
signal level. The on-to-off is the "Lower" value and the
off-to-on the "Upper." This implementation provides a hysteresis
detector. Early fldigi CW decoders used this scheme but the
detection levels were not adjustable.
<br>
<br>

The RxWPM control is an indicator and is not used for setting the operation
of the CW decoder.
<br>
<br>

"SOM decoding" provides a fuzzy logic implementation to match the RX
stream detected on-off sequence to a "best fit" character. It can
increase the probability of correctly identifying the text character
under very noisy conditions.
<br>

The TxWPM sliding controller is used to set the transmit WPM. To
make the setting easier two additional controls are provided.
"Lower" sets the lower limit of the slider and "Upper" sets the
upper limit of the slider. The resolution of the TxWPM slider is
1 WPM. The Lower/Upper controls are in in 5 WPM increments.
<br>
<br>

The transmit encoder settings for WPM can also be adjusted with three hot
keys:
<br>

<ul>
<li>Numeric keypad "+" increases the TxWPM by 1</li>
<li>Numeric keypad "-" decreases the TxWPM by 1</li>
<li>Numeric keypad "*" toggles between the selected TxWPM and a default WPM</li>
</ul>

The "Default" control on the CW tab sets that default value. As
shown above the TxWPM is 30 and the default is 18. If during a
QSO you needed to slow down to give the other op a better chance to
copy what you are sending, just hit the "*" on the numeric keypad and
the CW code will immediately switch to sending CW at the set default
value (18 wpm in this example). Press the "*" again to return to
back to the CW speed that you were previously using.<br>

\section subsec_timing Timing
<br>

\image html config-cw-qsk.png "Timing"
\image latex config-cw-qsk.png "Timing" width=5.0in
<br>
<br>

Fldigi generates CW by inserting a keyed tone at the current
waterfall audio frequency. The transceiver should be operated in
either USB (preferred) or LSB mode.
The CW signal is completely generated in the software so it is possible
to control many aspects of the CW signal. The actual transmitted
signal will be at the USB carrier + the audio frequency, or the LSB carrier -
the audio frequency. If fldigi is tracking and receiving a CW
signal on the waterfall your transmitted signal will be exactly on the
frequency of the other operator. The CW generated this way
has a nearly ideal attack and decay time, controlled by the software modem.
But ... a caveat ... your transmitter must never be
overdriven and it should have excellent opposite sideband suppression.
Overdriving the transmitter can cause multiple audio signals
within the SSB passband, and cause unwanted interference to other ops.
The same is true for a poorly designed or adjusted transmitter
with bad sideband suppression. I recommend having a trusted
and knowledgable operator assist you when first trying A2 CW.
Have them carefully look for evidence of your signal above and below your
primary signal (by at least +/- 3 Khz). If there is no
evidence of extra signals then your are set to go. If there is you
might want to have the transceiver adusted for sideband suppression, or check
to be sure you are not over driving the audio.

<ul>
<li>Wt % control sets the weight of the CW. Normal CW is at 50%
weight, ie: a dot is equal to the interval between dots or between code
elements. It has a range of 20 to 80 percent.</li>

<li>Dash/Dot controls the relative weight between a dash and a dot.
The standard for CW is 3 to 1. The dash is 3 times the
length of a dot. Some operators prefer the sound of either a
heavier or lighter sounding CW. This control can be adjusted from
2.5 to 4.0 in 0.1 increments.</li>

<li>Edge shape provides two leading/trailing edge shapes (1) Hanning, or raised
cosine, and (2) Blackman a modified raised cosine with a steeper attack
and decay. Both of these edge shapes give a more narrow bandwidth
CW signal than the traditional exponential waveform. They are
very easy to listen to even at speeds exceeding 100 wpm.</li>

<li>The Edge control sets the rise and fall times of the CW waveform.
It can be set anywhere from 0.0 to 15.0 milliseconds in
0.1 millisecond increments. DO NOT operate A2 CW with the control
set below 4 msec. This is the control that sets the effective
bandwidth and sound of your CW. If the edge is too steep you will
have a clicky signal and be the bane of the CW bands. The
purpose of being able to set the edge to 0.0 or a very quick rise/fall
time is explained below. A good setting for nice sounding CW at
40 WPM and below is 4 to 6 milliseconds.</li>

<li>Edge decreases pulse width, when checked will give a slightly narrower dot
length as the edge timing is increased. This is useful when
operating QSK and listening between the character elements.</li>
</ul>

This is what the A2 signal should look like with various settings of
weight, Dash/Dot and Edge. The audio frequency is 400 Hz and
the TxWPM is 100 WPM.
<br>
<br>

\image html cw-00-30.png "Dash/Dot = 3.0, Edge = 0.0"
\image latex cw-00-30.png "Dash/Dot = 3.0, Edge = 0.0" width=2.8in
<br>
\image html cw-00-40.png "Dash/Dot = 4.0, Edge = 0.0"
\image latex cw-00-40.png "Dash/Dot = 4.0, Edge = 0.0" width=2.8in
<br>
\image html cw-30-30.png "Dash/Dot = 3.0, Edge = 3 msec"
\image latex cw-30-30.png "Dash/Dot = 3.0, Edge = 3 msec" width=2.8in
<br>
\image html cw-60-30.png "Dash/Dot = 3.0, Edge = 6 msec"
\image latex cw-60-30.png "Dash/Dot = 3.0, Edge = 6 msec" width=2.8in
<br>
\image html cw-120-30.png "Dash/Dot = 3.0, Edge = 12 msec"
\image latex cw-120-30.png "Dash/Dot = 3.0, Edge = 12 msec" width=2.8in
<br>

Changing the weight, dash/dot or edge of the waveform does not change
the WPM at which the code is generated. When a conflict
occurs between the various settings WPM takes first priority, and Edge second.
In the above examples, the Edge setting could not exceed 12
msec even if the control were set higher than 12.0. The figures
were generated by capturing the output data being sent to the sound card and
then formatting it using Gnumeric. An oscilloscope photo of
the signal is virtually identical.<br>
<br>

The setting for inter-character and inter-word spacings are fixed at 3
and 7 respectively. The 3 is achieved by sending a silent
period of 1 dot (element) length at the beginning of each character and 2 at
the end of each character (shown in the figures). This silent
period is sufficient for most transceivers to respond to the PTT signal
which occurs at the beginning of the transmission so that the first dit
or dash is not lost in transmission.QRQ (high speed CW operation)<br>
<br>

You may wonder why fldigi can go as high as 200 WPM. It's hard to
believe but there are CW operators who can decode 100+ WPM in their
head. These operators also usually operate QSK (full
breakin). A2 CW and PTT operation and QRQ/QSK are not a natural mix.
But fldigi can be used for this type of operation if an
external keyer is used. For that purpose the A2 Tx output from fldigi
is full wave rectified and detected to create a keyline control.
The outboard conversion from A2 to keyline requires a nearly square wave
pulse output of audio at the CW keying rate. Setting the Edge
control to 0.0 and then the audio frequency to about 1000 Hz provides
the needed signal to effect this type of keyline control.<br>
<br>

If you are operating QSK with a separate receiver / transmitter you can
very quickly stop your transmit signal with the TAB key. In
the CW mode only the TAB key causes the program to skip over the remaining
text in the transmit text buffer. The text that is skipped
will be color coded blue. The program remains in the transmit mode
(PTT enabled), but since the buffer is now empty no A2 CW signal is
generated. Code transmission will then restart with the very
next keyboard closure of a valid CW character. The Escape and
Pause/Break keys still can be used to respectively abort and pause
transmission.
<br>

\section subsec_qsk QSK
<br>

You might ask why fldigi doesn't simply provide a keyline output on
one of the parallel port pins or on RTS or DTR via a comm port.
The answer is quite simple. Linux is a multi-tasking
operating system and the interaction between the OS and the application
causes the timing to be adversely effected.  The driver
implementation of the audio sub system must be responsive and so the OS
gives that sub system a very high priority in its multi-tasking
structure.
<br>
<br>

Many QSK operators use high speed diode antenna switching between
receiver and antenna. fldigi generates a signal that can be used
for that purpose. The left audio channel is always theAFCW
signal. When selected the right audio channel can be configured to
generate a square wave signal that begins earlier and ends later than each
of the CW elements. The square wave signal can be rectified and filtered
to provide the diode switching signal for the Rx/Tx antenna switching.<br>
<br>

The right audio channel QSK signal is selected by checking the box and
then adjusting the pre and post timing in millisecond increments.
Additional information and a schematic diagram of a QSK keying circuit
is described in \ref cw_keying_page "CW Keying". Setting
up a QSK device can be quite difficult. Fldigi helps to ease the
adjustment by generating a continuous series of characters. This
allows a dual trace scope to be properly synched while making the
adjustments to both the software and the associated QSK hardware.
You enable continuous characters by selecting the checkbox, and
then enabling the T/R button for transmit. The repeated character
can be change on the fly with the pick control. It can be one of
either E, I, S, T, M, O or V.
<br>
<br>

\section subsec_prosigns PROSIGNS

\image html config-cw-prosigns.png "CW Prosigns"
\image latex config-cw-prosigns.png "CW Prosigns" width=5.7in
<br>
\image html config-cw-prosign-popup.png "Prosign Popup"
\image latex config-cw-prosign-popup.png "Prosign Popup" width=1.0in

You can assign keyboard characters to be used for Morse prosigns. The
available characters are: ~ % &amp; + = { } &lt; &gt; [ ]
<br>
<br>

The default assignments are shown above. You can also elect to send and
receive the KN prosign as an open parenthesis '('. This is commonly
used on MARS CW operations.
<br>
<br>

See \ref cw_page "Operating CW" for additional information.
<br>
<br>

\ref cw_configuration_page "Return to Top of Page"
<br>
\ref main_page "Return to Main Page"


*/


NEWS
= Fldigi 3.21 release notes =

== Modems ==

  * Weather Fax receive and transmit
  * RSID can be configured to open the decoder squelch
  * Added several new RSID modem codes
  * Video ID improvements
  * Added browser support for the PSK63F and PSK-nnn-R types
  * Improved CW prosign handling
  * Improved signal detection in PSK multiple decoder
  * The MT-63 modem can now receive and transmit on any frequency
  * Added support for multiple RTTY decoders

== Macros ==

  * Added REPEAT macro
  * Added LNW macro
  * Added CLEARLOG macro
  * Added QSY macro
  * Added RIGMOD and FILWID macros

== User interface ==

  * Added Italian and Spanish translations
  * Translations now also work on win32
  * Added second (optional) macro bar
  * Improved PSK/RTTY browser widget

== Other changes ==

  * RigCAT can use RTS and DTR PTT simultaneously
  * Added interface to Logbook server
  * Added support for ADIF 2.2.3 QSO_DATE_OFF field
  * Added support for CALLOOK (http://callook.info)

== Bug fixes ==

  * Yes


= Fldigi 3.20 release notes =

== Modems ==

 * Added new PSK-500 mode
 * Added "robust PSK" modes: PSK-125R, PSK-250R, PSK-500R
 * Added PSK63F mode, compatible with MultiPsk PSK63FEC
 * Added Contestia support
 * Added SSB "modem" to simplify logging of SSB QSOs
 * Added MT63 modem preamble tones to aid signal acquisition
 * Added selectable filter widths for Hellschreiber modes
 * Added AGC and S/N detector for FM-HELL modes
 * Added Olivia noise estimator

 * Improved THOR and DominoEX squelch responsiveness
 * Improved Olivia squelch detection
 * Improved Feldhell AGC and S/N detector
 * Improved RTTY decoder

 * Removed THOR soft decoding
 * RSID now works at the same time as modem decoding
 * The HELL-80 tone spacing was changed to 300 Hz

== Macros ==

 * Added <LOCK:on|off|t> macro to enable/disable/toggle the transmit lock
 * Added <TUNE:nn> macro to transmit tuning tone for ''nn'' seconds
 * Added <RXRSID:on|off|t> and <TXRSID:on|off|t> macros
 * Added <WAIT:nn> macro to delay macro expansion by ''nn'' seconds
 * Added <SRCHUP>, <SRCHDN>, <GOHOME>, <GOFREQ:nnnn> macros to search for a
   signal up and down in frequency, go to the "sweet spot frequency", and
   set the current modem frequency to ''nnnn''
 * The <MODEM> macro syntax is now <MODEM:str>. The older <MODEM>str syntax is
   still supported.  Modem parameters may be specified when using the new
   syntax; currently for RTTY (shift, baud, bits), Olivia (bw, tones) and
   Contestia (bw, tones), e.g.: <MODEM:RTTY:170:45.45:5>, <MODEM:OLIVIA:500:16>,
   <MODEM:CTSTIA:250:8>.

== User interface ==

 * The main window's QSO and waterfall toolbar are now much more configurable
 * The QSO toolbar can now be set to Contest mode, Minimal mode, Full mode,
   and some combinations thereof
 * The docked digiscope can be hidden and shown without a restart
 * It is now possible to select the modes for which to transmit RSID, VideoID
   and CWID
 * The Op Modes menu can now be customised
 * Removed the floating rig control window
 * Removed various Alt-<key> shortcuts
 * Added a simple "wizard" for first-time users
 * Added a waterfall-only mode; enabled with --wfall-only
 * Removed option to use check buttons for AFC and SQL toggles

== Other changes ==

 * The XML-RPC server now supports HTTP/1.1 keep-alive connections
 * Added several new XML-RPC methods
 * Renamed various main.set_rig_NNN XML-RPC methods to rig.set_NNN
 * Removed obsolete command line switches: --twoscopes, --experimental
 * Improved support for PulseAudio, now also the default backend on Linux
 * The ^R string is no longer treated like ^r; it is now ignored unless at the
   end of the text to be transmitted
 * The Portaudio playback stream is no longer kept open in receive mode; this
   fixes a problem with certain rig interfaces that generate a PTT signal even
   when the audio is complete silence
 * Improved win32 serial port support

== Bug fixes ==

 * The MT63 transmit buffer is now correctly initialized
 * Fixed random exceptions-related crashes on win32
 * Stabilized AFC for all PSK baud rates
 * Improved sensitivity of RSID implementation

 Many more small (and a couple not so small) bug fixes all over the tree


src/config.h
// ----------------------------------------------------------------------------
// config.h  --  user configuration items for fldigi
//
// Copyright (C) 2006
//		Dave Freese, W1HKJ
//
// This file is part of fldigi.  Adapted in part from code contained in gmfsk 
// source code distribution.
//
// fldigi is free software; you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation; either version 2 of the License, or
// (at your option) any later version.
//
// fldigi is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with fldigi; if not, write to the Free Software
// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
// ----------------------------------------------------------------------------

#ifndef _CONFIG_H
#define _CONFIG_H

// You can change the x1 width of the waterfall / spectrum display by modifying this
// constant.
// Suggest that you make the value a multiple of 100.
// DO NOT EXCEED 4000
// The larger the number the greater the cpu load will be for creating the
// waterfall display
#define DEFAULT_IMAGE_WIDTH 3000
#define DEFAULT_HWFALL 140
#define DEFAULT_HNOM 570
#define DEFAULT_WNOM (DEFAULT_IMAGE_WIDTH / 4 + DEFAULT_HWFALL - BTN_HEIGHT)

#define EMC_HWFALL 120
#define EMC_HNOM 400
#define EMC_WNOM (EMC_HWFALL + 580)

extern int IMAGE_WIDTH;
extern int Hwfall;
extern int HNOM;
extern int WNOM;

// widget sizes internal to the waterfall widget
#define BEZEL		 2
#define WFTEXT		10
#define WFSCALE     10
#define WFMARKER     6
#define BTN_HEIGHT	20

#define bwColor		30
#define bwFFT		30
#define bwX1		30
#define bwMov		20
#define cwCnt		100
#define cwRef		50
#define cwMode		85
#define bwQsy		45
#define bwRate		45
#define bwXmtLock	45
#define bwRev		45
#define bwXmtRcv	45
#define wSpace		2

#define Hmenu		22
#define Hqsoframe	48
#define Hnotes		22
#define Hstatus		22
#define Hmacros		22

#define Htext		(HNOM - 4 - Hwfall - Hmenu - Hstatus - Hmacros - Hqsoframe - Hnotes)
// Htext = HNOM - 140 - Hwfall
#define Hrcvtxt		(Htext) / 2
#define Hxmttxt		(Htext - (Hrcvtxt))

#define Wwfall		(WNOM - Hwfall + BTN_HEIGHT + 2 * BEZEL)
#define Wmode 		80
#define Ws2n  		100
#define Wimd  		100
#define Wwarn 		16
#define bwAfcOnOff	(Hwfall -22)/2
#define bwSqlOnOff	(Hwfall -22)/2

#define Wstatus (WNOM - Wmode - Ws2n - Wimd - bwAfcOnOff - bwSqlOnOff - Wwarn)

//remove the comment delimiter to enable experimental psk250 and qpsk250 modes

#define USE250

//remove the comment delimiter to enable both old and new text widget class
//switching from one to the other requires a restart of the program before
//the newly selected widget becomes effective.
//
//default is to use the new text widget class that is a fully enabled editor
//widget
//#define USE_BOTH_TEXT_WIDGETS


#endif
